doc/Mutable_References.md

```diff
@@ -330,6 +331,18 @@ is effectively a wrapper for a `Cell` that contains an `Option`, like:
 `None` if the value is effectively unset. Using an `OptionalCell` (like a
 `NumCell`) makes the code clearer and hides extra tedious function calls.
 
+### Comparison to `TakeCell`
+
+`TakeCell` and `OptionalCell` are quite similar, but the key differentiator is
+the `Copy` bound for items stored in `OptionalCell`s. The semantic difference
```

Comment:
```suggestion
the `Copy` bound required for items to use some of the methods defined on `OptionalCell`, such as `map()`.
The `Copy` bound enables safe "reentrant" access to the stored value, because multiple accesses will be operating
on a copy of the stored item. The semantic difference
```

I want it to be clear that non-Copy items *can* be stored in an `OptionalCell`, but `map()` etc. cannot be called on those `OptionalCell`s.